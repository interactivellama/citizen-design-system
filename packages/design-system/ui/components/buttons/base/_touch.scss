.slds-button {
  @include border-box-line-height($height-tappable);
  font-weight: $font-weight-bold;
}

.slds-button__icon {
  @include square($square-icon-medium-content);
}
