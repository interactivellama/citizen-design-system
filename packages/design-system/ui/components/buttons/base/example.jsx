// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import classNames from 'classnames';

export let Button = props => {
  const {
    className,
    disabled,
    isBrand,
    isNeutral,
    isOutlineBrand,
    isInverse,
    isDestructive,
    isTextDestructive,
    isSuccess,
    isFullWidth,
    isStretch,
    isFirst,
    isMiddle,
    isLast,
    ...rest
  } = props;

  return (
    <button
      className={classNames(
        'slds-button',
        {
          'slds-button_brand': isBrand,
          'slds-button_neutral': isNeutral,
          'slds-button_outline-brand': isOutlineBrand,
          'slds-button_inverse': isInverse,
          'slds-button_destructive': isDestructive,
          'slds-button_text-destructive': isTextDestructive,
          'slds-button_success': isSuccess,
          'slds-button_full-width': isFullWidth,
          'slds-button_stretch': isStretch,
          'slds-button_first': isFirst,
          'slds-button_middle': isMiddle,
          'slds-button_last': isLast
        },
        className
      )}
      disabled={disabled}
      {...rest}
    >
      {props.children}
    </button>
  );
};

/// ////////////////////////////////////////
// Export
/// ////////////////////////////////////////

export default <Button>Button</Button>;

export let examples = [
  {
    id: 'neutral',
    label: 'Neutral',
    element: <Button isNeutral>Neutral Button</Button>
  },
  {
    id: 'brand',
    label: 'Brand',
    element: <Button isBrand>Brand Button</Button>
  },
  {
    id: 'outline-brand',
    label: 'Outline Brand',
    element: <Button isOutlineBrand>Outline Brand Button</Button>
  },
  {
    id: 'inverse',
    label: 'Inverse',
    element: (
      <div style={{ backgroundColor: '#16325c', padding: '0.5rem' }}>
        <Button isInverse>Inverse Button</Button>
      </div>
    )
  },
  {
    id: 'destructive',
    label: 'Destructive',
    element: <Button isDestructive>Destructive Button</Button>
  },
  {
    id: 'text-destructive',
    label: 'Text Destructive',
    element: <Button isTextDestructive>Text Destructive Button</Button>
  },
  {
    id: 'success',
    label: 'Success',
    element: <Button isSuccess>Success Button</Button>
  },
  {
    id: 'stretch',
    label: 'Stretch',
    element: (
      <Button isStretch isNeutral>
        Stretched Neutral Button
      </Button>
    )
  }
];

export let states = [
  {
    id: 'disabled',
    label: 'Disabled',
    element: <Button disabled>Button</Button>
  },
  {
    id: 'neutral-disabled',
    label: 'Neutral - Disabled',
    element: (
      <Button isNeutral disabled>
        Neutral Button
      </Button>
    )
  },
  {
    id: 'brand-disabled',
    label: 'Brand - Disabled',
    element: (
      <Button isBrand disabled>
        Brand Button
      </Button>
    )
  },
  {
    id: 'outline-brand-disabled',
    label: 'Outline Brand - Disabled',
    element: (
      <Button isOutlineBrand disabled>
        Outline Brand Button
      </Button>
    )
  },
  {
    id: 'inverse-disabled',
    label: 'Inverse - Disabled',
    element: (
      <div style={{ backgroundColor: '#16325c', padding: '0.5rem' }}>
        <Button isInverse disabled>
          Inverse Button
        </Button>
      </div>
    )
  },
  {
    id: 'destructive-disabled',
    label: 'Destructive - Disabled',
    element: (
      <Button isDestructive disabled>
        Destructive Button
      </Button>
    )
  },
  {
    id: 'text-destructive-disabled',
    label: 'Text Destructive - Disabled',
    element: (
      <Button isTextDestructive disabled>
        Text Destructive Button
      </Button>
    )
  },
  {
    id: 'success-disabled',
    label: 'Success - Disabled',
    element: (
      <Button isSuccess disabled>
        Success Button
      </Button>
    )
  }
];
