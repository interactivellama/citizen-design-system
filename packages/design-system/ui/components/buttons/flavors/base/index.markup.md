The base `.{{cssPrefix}}button` looks like a plain text link. It removes all the styling of the native button. It’s typically used to trigger a modal or display a “like” link. All button variations are built by adding another class to `.{{cssPrefix}}button`.

Add the `.{{cssPrefix}}button--neutral` class to create a neutral button, which has a white background and gray border.

Use a neutral icon button is for buttons with an icon on the left or right (not for stateful buttons). Add the `.{{cssPrefix}}button--neutral` class to `.{{cssPrefix}}button`.

The SVG inside receives the `.{{cssPrefix}}button__icon` class. You can position the icon on the right or the left using `.{{cssPrefix}}button__icon--right` or `.{{cssPrefix}}button__icon--left` , which apply the correct amount of space between the icon and the text.

To create the brand button, add the `.{{cssPrefix}}button--brand` class to the `.{{cssPrefix}}button` class.

To create the destructive button, add the `.{{cssPrefix}}button--destructive` class to the `.{{cssPrefix}}button` class.

Use the inverse button on dark backgrounds. Add the `.{{cssPrefix}}button--inverse` class to the `.{{cssPrefix}}button` class.
