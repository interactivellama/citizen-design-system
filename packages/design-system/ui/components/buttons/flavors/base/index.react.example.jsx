// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import SvgIcon from 'app_modules/ui/svg-icon';
import classNames from 'classnames';

///////////////////////////////////////////
// Partial(s)
///////////////////////////////////////////

let Demo = props =>
  <div className="demo-only" {...props}>
    {props.children}
  </div>;

let Button = props =>
  <button className={classNames('slds-button', props.className)} disabled={props.disabled}>
    {props.children}
  </button>;

///////////////////////////////////////////
// State Constructor(s)
///////////////////////////////////////////

let Default = props =>
  <Demo>
    <Button>Button</Button>
  </Demo>;

let Neutral = props =>
  <Demo>
    <Button className="slds-button--neutral">Button Neutral</Button>
  </Demo>;

let NeutralIconLeft = props =>
  <Demo>
    <Button className="slds-button--neutral">
      <SvgIcon className="slds-button__icon slds-button__icon--left" sprite="utility" symbol="download" />
      Button Neutral
    </Button>
  </Demo>;

let NeutralIconRight = props =>
  <Demo>
    <Button className="slds-button--neutral">
      Button Neutral
      <SvgIcon className="slds-button__icon slds-button__icon--right" sprite="utility" symbol="down" />
    </Button>
  </Demo>;

let NeutralDisabled = props =>
  <Demo>
    <Button className="slds-button--neutral" disabled>Button Neutral</Button>
  </Demo>;

let Brand = props =>
  <Demo>
    <Button className="slds-button--brand">Button Brand</Button>
  </Demo>;

let BrandDisabled = props =>
  <Demo>
    <Button className="slds-button--brand" disabled>Button Brand</Button>
  </Demo>;

let Destructive = props =>
  <Demo>
    <Button className="slds-button--destructive">Button Destructive</Button>
  </Demo>;

let DestructiveDisabled = props =>
  <Demo>
    <Button className="slds-button--destructive" disabled>Button Destructive</Button>
  </Demo>;

let Inverse = props =>
  <Demo style={{ padding: '0.5rem', background: '#16325c' }}>
    <Button className="slds-button--inverse">Button Inverse</Button>
  </Demo>;

let InverseDisabled = props =>
  <Demo style={{ padding: '0.5rem', background: '#16325c' }}>
    <Button className="slds-button--inverse" disabled>Button Inverse</Button>
  </Demo>;

///////////////////////////////////////////
// Export
///////////////////////////////////////////

export let states = [
  {
    id: 'button',
    label: 'Default',
    element: <Default />
  },
  {
    id: 'button-neutral',
    label: 'Neutral',
    element: <Neutral />
  },
  {
    id: 'button-neutral-disabled',
    label: 'Neutral disabled',
    element: <NeutralDisabled />
  },
  {
    id: 'button-neutral-icon-left',
    label: 'Neutral with left icon',
    element: <NeutralIconLeft />
  },
  {
    id: 'button-neutral-icon-right',
    label: 'Neutral with right icon',
    element: <NeutralIconRight />
  },
  {
    id: 'button-brand',
    label: 'Brand',
    element: <Brand />
  },
  {
    id: 'button-brand-disabled',
    label: 'Brand disabled',
    element: <BrandDisabled />
  },
  {
    id: 'button-destructive',
    label: 'Destructive',
    element: <Destructive />
  },
  {
    id: 'button-destructive-disabled',
    label: 'Destructive disabled',
    element: <DestructiveDisabled />
  },
  {
    id: 'button-inverse',
    label: 'Inverse',
    element: <Inverse />
  },
  {
    id: 'button-inverse-disabled',
    label: 'Inverse disabled',
    element: <InverseDisabled />
  },
  {
    id: 'success',
    label: 'Success',
    element: <Button className="slds-button--success">Button Success</Button>
  }
];
