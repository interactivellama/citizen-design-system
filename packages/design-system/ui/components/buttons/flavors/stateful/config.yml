status: 'dev-ready'
dependencies:
  - components/buttons
  - components/buttons/flavors/neutral
  - components/buttons/flavors/icon
  - components/buttons/flavors/neutral-icon
  - core/icons
showFormFactors:
  - large
