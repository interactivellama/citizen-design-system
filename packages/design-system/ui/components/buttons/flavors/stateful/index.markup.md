The stateful button requires the `.{{cssPrefix}}button--neutral` class in addition to the `.{{cssPrefix}}button` class.

It uses the class `.{{cssPrefix}}not-selected` in its initial state. When the user activates the button, use JavaScript to toggle the class to `.{{cssPrefix}}is-selected`. The button contains three spans with classes that hide or show the content of the spans based on the class on the button. Each span contains text and a corresponding icon. The svg will have the `.{{cssPrefix}}button__icon--stateful` class as well as the `.{{cssPrefix}}button__icon--left` class setting the icon on the left.

<h4 class="site-text-heading--label">Accessibility</h4>

For accessibility, include the attribute `aria-live="assertive"` on the button. The <code>aria-live="assertive"</code> attribute means the value of the <code>&lt;span&gt;</code> inside the button will be spoken whenever it changes.

