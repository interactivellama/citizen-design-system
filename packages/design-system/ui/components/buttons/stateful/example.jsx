// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import SvgIcon from 'app_modules/ui/svg-icon';
import classNames from 'classnames';

let StatefulButton = props =>
  <button
    className={classNames('slds-button slds-button--stateful', props.className)}
    disabled={props.disabled}
    aria-live="assertive"
  >
    <span className="slds-text-not-selected">
      <SvgIcon
        className="slds-button__icon--stateful slds-button__icon--left"
        sprite="utility"
        symbol="add"
      />
      Follow
    </span>
    <span className="slds-text-selected">
      <SvgIcon
        className="slds-button__icon--stateful slds-button__icon--left"
        sprite="utility"
        symbol="check"
      />
      Following
    </span>
    <span className="slds-text-selected-focus">
      <SvgIcon
        className="slds-button__icon--stateful slds-button__icon--left"
        sprite="utility"
        symbol="close"
      />
      Unfollow
    </span>
  </button>;

///////////////////////////////////////////
// Export
///////////////////////////////////////////

export default (
  <StatefulButton className="slds-button--neutral slds-not-selected" />
);
