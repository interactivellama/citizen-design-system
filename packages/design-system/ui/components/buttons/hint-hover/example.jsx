// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import SvgIcon from 'app_modules/ui/svg-icon';
import classNames from 'classnames';
import { Button } from 'ui/components/buttons/base/example';

///////////////////////////////////////////
// Export
///////////////////////////////////////////

export default (
  <div className="slds-hint-parent">
    <Button className="slds-button slds-button--hint">Button</Button>
  </div>
);
