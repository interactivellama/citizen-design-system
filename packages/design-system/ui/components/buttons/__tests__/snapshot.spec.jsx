/* eslint-env jest */
import React from 'react';
import { Button } from '../base/example';
import { StatefulButton } from '../stateful/example';
import createHelpers from '../../../../jest.helpers';

const { matchesMarkup } = createHelpers(__dirname);

describe('Button ', () => {
  it('renders a base button', () => matchesMarkup(<Button>Button</Button>));

  it('renders a disabled base button', () =>
    matchesMarkup(<Button disabled>Button</Button>));

  it('renders a neutral button', () =>
    matchesMarkup(<Button isNeutral>Button</Button>));

  it('renders a brand button', () =>
    matchesMarkup(<Button isBrand>Button</Button>));

  it('renders an outline brand button', () =>
    matchesMarkup(<Button isOutlineBrand>Button</Button>));

  it('renders an inverse button', () =>
    matchesMarkup(<Button isInverse>Button</Button>));

  it('renders a desctructive button', () =>
    matchesMarkup(<Button isDestructive>Button</Button>));

  it('renders a text destructive button', () =>
    matchesMarkup(<Button isTextDestructive>Button</Button>));

  it('renders a success button', () =>
    matchesMarkup(<Button isSuccess>Button</Button>));
});

describe('Stateful Button ', () => {
  it('renders a not-selected stateful button', () =>
    matchesMarkup(<StatefulButton isNotSelected>Button</StatefulButton>));

  it('renders a selected and focused stateful button', () =>
    matchesMarkup(<StatefulButton isSelectedClicked>Button</StatefulButton>));

  it('renders a selected stateful button', () =>
    matchesMarkup(<StatefulButton isSelected>Button</StatefulButton>));
});
