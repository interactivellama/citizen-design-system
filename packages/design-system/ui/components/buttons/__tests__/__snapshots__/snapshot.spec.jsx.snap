// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button  renders a base button 1`] = `
<button
  className="slds-button"
  disabled={undefined}
>
  Button
</button>
`;

exports[`Button  renders a brand button 1`] = `
<button
  className="slds-button slds-button_brand"
  disabled={undefined}
>
  Button
</button>
`;

exports[`Button  renders a desctructive button 1`] = `
<button
  className="slds-button slds-button_destructive"
  disabled={undefined}
>
  Button
</button>
`;

exports[`Button  renders a disabled base button 1`] = `
<button
  className="slds-button"
  disabled={true}
>
  Button
</button>
`;

exports[`Button  renders a full width, reset button 1`] = `
<button
  className="slds-button slds-button_full-width"
  disabled={undefined}
>
  Button
</button>
`;

exports[`Button  renders a neutral button 1`] = `
<button
  className="slds-button slds-button_neutral"
  disabled={undefined}
>
  Button
</button>
`;

exports[`Button  renders a stretched button 1`] = `
<button
  className="slds-button slds-button_success slds-button_stretch"
  disabled={undefined}
>
  Button
</button>
`;

exports[`Button  renders a success button 1`] = `
<button
  className="slds-button slds-button_success"
  disabled={undefined}
>
  Button
</button>
`;

exports[`Button  renders a text destructive button 1`] = `
<button
  className="slds-button slds-button_text-destructive"
  disabled={undefined}
>
  Button
</button>
`;

exports[`Button  renders an inverse button 1`] = `
<button
  className="slds-button slds-button_inverse"
  disabled={undefined}
>
  Button
</button>
`;

exports[`Button  renders an outline brand button 1`] = `
<button
  className="slds-button slds-button_outline-brand"
  disabled={undefined}
>
  Button
</button>
`;

exports[`Dual Stateful Button renders a disabled dual stateful button 1`] = `
<button
  aria-live="assertive"
  className="slds-button slds-button_neutral slds-button_dual-stateful"
  disabled={true}
>
  <span
    className="slds-text-not-pressed"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#add"
      />
    </svg>
    Follow
  </span>
  <span
    className="slds-text-pressed"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
      />
    </svg>
    Following
  </span>
</button>
`;

exports[`Dual Stateful Button renders a not-pressed dual stateful button 1`] = `
<button
  aria-live="assertive"
  className="slds-button slds-button_neutral slds-button_dual-stateful"
  disabled={undefined}
>
  <span
    className="slds-text-not-pressed"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#add"
      />
    </svg>
    Follow
  </span>
  <span
    className="slds-text-pressed"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
      />
    </svg>
    Following
  </span>
</button>
`;

exports[`Dual Stateful Button renders a pressed dual stateful button 1`] = `
<button
  aria-live="assertive"
  className="slds-button slds-button_neutral slds-button_dual-stateful slds-is-pressed"
  disabled={undefined}
>
  <span
    className="slds-text-not-pressed"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#add"
      />
    </svg>
    Follow
  </span>
  <span
    className="slds-text-pressed"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
      />
    </svg>
    Following
  </span>
</button>
`;

exports[`Stateful Button  renders a not-selected stateful button 1`] = `
<button
  aria-live="assertive"
  className="slds-button slds-button_neutral slds-button_stateful slds-not-selected"
  disabled={undefined}
>
  <span
    className="slds-text-not-selected"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#add"
      />
    </svg>
    Follow
  </span>
  <span
    className="slds-text-selected"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
      />
    </svg>
    Following
  </span>
  <span
    className="slds-text-selected-focus"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
      />
    </svg>
    Unfollow
  </span>
</button>
`;

exports[`Stateful Button  renders a selected and focused stateful button 1`] = `
<button
  aria-live="assertive"
  className="slds-button slds-button_neutral slds-button_stateful slds-is-selected-clicked"
  disabled={undefined}
>
  <span
    className="slds-text-not-selected"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#add"
      />
    </svg>
    Follow
  </span>
  <span
    className="slds-text-selected"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
      />
    </svg>
    Following
  </span>
  <span
    className="slds-text-selected-focus"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
      />
    </svg>
    Unfollow
  </span>
</button>
`;

exports[`Stateful Button  renders a selected stateful button 1`] = `
<button
  aria-live="assertive"
  className="slds-button slds-button_neutral slds-button_stateful slds-is-selected"
  disabled={undefined}
>
  <span
    className="slds-text-not-selected"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#add"
      />
    </svg>
    Follow
  </span>
  <span
    className="slds-text-selected"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#check"
      />
    </svg>
    Following
  </span>
  <span
    className="slds-text-selected-focus"
  >
    <svg
      aria-hidden={true}
      className="slds-button__icon slds-button__icon_small slds-button__icon_left"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
      />
    </svg>
    Unfollow
  </span>
</button>
`;
